plugin/src/main/java/com/playmonumenta/scriptedquests/zones/ZoneLayer.java

```diff
@@ -0,0 +1,365 @@
+
+	private boolean hasPropertyGroupLoop(HashMap<String, LinkedHashSet<String>> groupReferences, String startGroup) {
+		return hasPropertyGroupLoop(groupReferences, startGroup, startGroup);
+	}
+
+	private boolean hasPropertyGroupLoop(HashMap<String, LinkedHashSet<String>> groupReferences, String startGroup, String continueGroup) {
+		LinkedHashSet<String> subGroupReferences = groupReferences.get(continueGroup);
+		if (subGroupReferences == null) {
+			return false;
+		}
+
+		for (String subGroupReference : subGroupReferences) {
+			if (subGroupReference.equals(startGroup)) {
+				return true;
+			}
+
+			if (hasPropertyGroupLoop(groupReferences, startGroup, subGroupReference)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/*
+	 * This should only be called by the ZoneManager.
+	 */
+	public static void clearDynmapLayers() {
+		DynmapCommonAPI dynmapHook = (DynmapCommonAPI) Bukkit.getServer().getPluginManager().getPlugin("dynmap");
+		if (dynmapHook == null) {
+			return;
+		}
+
+		MarkerAPI markerHook = dynmapHook.getMarkerAPI();
+		if (markerHook == null) {
+			// Not initialized
+			return;
+		}
+
+		// Duplicate set of MarkerSets in case the return value is a view, not a copy.
+		LinkedHashSet<MarkerSet> allMarkerSets = new LinkedHashSet<MarkerSet>();
+		allMarkerSets.addAll(markerHook.getMarkerSets());
+		for (MarkerSet markerSet : allMarkerSets) {
+			if (markerSet != null &&
+			    markerSet.getMarkerSetID().startsWith(DYNMAP_PREFIX) &&
+			    !markerSet.isMarkerSetPersistent()) {
+				markerSet.deleteMarkerSet();
+			}
+		}
+	}
+
+	private void refreshDynmapLayer() {
+		DynmapCommonAPI dynmapHook = (DynmapCommonAPI) Bukkit.getServer().getPluginManager().getPlugin("dynmap");
+		if (dynmapHook == null) {
+			return;
+		}
+
+		MarkerAPI markerHook = dynmapHook.getMarkerAPI();
+		if (markerHook == null) {
+			// Not initialized
+			return;
+		}
+
+		String markerSetId = DYNMAP_PREFIX + mName.replace(" ", "_");
+		MarkerSet markerSet;
+
+		markerSet = markerHook.getMarkerSet(markerSetId);
+		if (markerSet != null) {
+			// Delete old marker set
+			markerSet.deleteMarkerSet();
+		}
+		// Create a new marker set
+		markerSet = markerHook.createMarkerSet(markerSetId, mName, null, false);
+
+		// Zones reversed so clicking on the overlap of two zones returns the highest priority zone.
+		// This isn't 100% consistent either way, but it's more consistent like this without needing
+		// to render every zone fragment (which is also an option).
+		for (int zoneIndex = mZones.size() - 1; zoneIndex >= 0; zoneIndex--) {
+			Zone zone = mZones.get(zoneIndex);
+
+			String zoneLabel = zone.getName();
+			int zoneColor = ZoneUtils.getColor(mName, zoneLabel);
+			String zoneId = zoneLabel.replace(" ", "_");
+
+			// TODO Replace the first false in createAreaMarker with true and make zoneLabel HTML markup here.
+
+			// TODO get the world string properly, not some hacked together solution
+			String world = Bukkit.getWorlds().get(0).getName();
```

Comment:
I'd remove this TODO comment - if we need to support per-world based zones (or anything for that matter) we're going to need to do so much stuff that this isn't going to be a helpful reminder. And if we don't move to a per-world based scheme, this will never happen and is just clutter.